cassiopeia/data/CassiopeiaTree.py

```diff
@@ -0,0 +1,185 @@
+"""
+This file stores the basic data structure for Cassiopeia - the
+CassiopeiaTree. At a minimum, this data structure will contain a character
+matrix containing that character state information for all the cells in a given
+clonal population. Other important data is also stored here, like the priors
+for given character states as well any meta data associated with this clonal 
+population.
+
+When a solver has been called on this object, a tree 
+will be added to the data structure at which point basic properties can be 
+queried like the average tree depth or agreement between character states and
+phylogeny.
+
+This object can be passed to any CassiopeiaSolver subclass as well as any
+analysis module, like a branch length estimator or rate matrix estimator
+"""
+import ete3
+import networkx as nx
+import pandas as pd
+from typing import Dict, List, Optional, Union
+
+from cassiopeia.data import utilities
+
+
+class CassiopeiaTree:
+    def __init__(
+        self,
+        character_matrix: pd.DataFrame,
+        cell_meta: Optional[pd.DataFrame] = None,
+        character_meta: Optional[pd.DataFrame] = None,
+        priors: Optional[Dict[int, Dict[int, float]]] = None,
+        tree: Optional[Union[str, ete3.Tree, nx.DiGraph]] = None,
+    ):
+
+        self.character_matrix = character_matrix
+        self.cell_meta = cell_meta
+        self.character_meta = character_meta
+        self.priors = priors
+        self.__network = None
```

Comment:
Can you give an idea of what cell metadata and character metadata might look like? Also, I can imagine it being useful to store some experiment metadata too, like the length of the experiment that might be useful in rescaling branch lengths

Reply:
Good comments! 

I've added descriptions of common cell and character meta data to the docstring.

There are indeed some extraneous experimental meta data that we can add, but I'm going to table this for now (a todo is left in the docstring). These will be good to converge on when we have distinct use-cases.

Reply:
Indeed. Thanks Matt!